api_tests/registrations/views/test_registration_list.py

```diff
@@ -837,6 +840,217 @@ def test_cannot_register_draft_that_has_already_been_approved(self, app, user, p
         assert res.json['errors'][0]['detail'] == 'This draft has already been approved and cannot be modified.'
 
 
+@pytest.mark.django_db
+class TestRegistrationBulkUpdate:
+
+    @pytest.fixture()
+    def url(self):
+        return '/{}registrations/'.format(API_BASE)
+
+    @pytest.fixture()
+    def user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def registration_one(self, user):
+        return RegistrationFactory(creator=user, embargo=EmbargoFactory(user=user), is_public=False)
+
+    @pytest.fixture()
+    def registration_two(self, user):
+        return RegistrationFactory(creator=user, embargo=EmbargoFactory(user=user), is_public=False)
+
+    @pytest.fixture()
+    def private_payload(self, registration_one, registration_two):
+        return {
+            'data': [
+                {
+                    'id': registration_one._id,
+                    'type': 'registrations',
+                    'attributes': {
+                        'public': False
+                    }
+                },
+                {
+                    'id': registration_two._id,
+                    'type': 'registrations',
+                    'attributes': {
+                        'public': False
+                    }
+                }
+            ]
+        }
+
+    @pytest.fixture()
+    def public_payload(self, registration_one, registration_two):
+        return {
+            'data': [
+                {
+                    'id': registration_one._id,
+                    'type': 'registrations',
+                    'attributes': {
+                        'public': True
+                    }
+                },
+                {
+                    'id': registration_two._id,
+                    'type': 'registrations',
+                    'attributes': {
+                        'public': True
+                    }
+                }
+            ]
+        }
+
+    @pytest.fixture()
+    def empty_payload(self, registration_one, registration_two):
+        return {
+            'data': [
+                {
+                    'id': registration_one._id,
+                    'type': 'registrations',
+                    'attributes': {}
+                },
+                {
+                    'id': registration_two._id,
+                    'type': 'registrations',
+                    'attributes': {}
+                }
+            ]
+        }
+
+    @pytest.fixture()
+    def bad_payload(self, registration_one, registration_two):
+        return {
+            'data': [
+                {
+                    'id': registration_one._id,
+                    'type': 'registrations',
+                    'attributes': {
+                        'title': 'Nerds',
```

Comment:
I would say do one valid first and an invalid second, so make this set the public while the next one sets the title. This should verify that a) the title didn't get set, and b) that the privacy setting did not change.